routes/adminAccess.js

```diff
@@ -0,0 +1,49 @@
+var express = require('express');
+var router = express.Router();
+var userMapper = require('../mappers/userMapper');
+var isLoggedIn = require('../config/utils').isLoggedIn;
+var isAdminUser = require('../config/utils').isAdminUser;
+
+router.get('/admin',isLoggedIn, isAdminUser, function(req, res) {
+	res.render('adminAccess', { user: req.user,err: req.flash('err'),succ: req.flash('succ')});
+});
+
+router.post('/admin', isLoggedIn, isAdminUser, function(req, res) {
+	if(validEmail(req.body)){
+		userMapper.findUserByEmail(req.body.userEmail, function(error, result) {
+			if (!result) {
+				req.flash('err', 'User email not found');
+				res.redirect('/');
+			} else if (error) {
+				req.flash('err', error);
+			} else {
+
+				var adminType = 'admin';
+				var adminGrantedUser =  {name: result[0].name, email: result[0].email, password:result[0].password, type: adminType, salt:result[0].salt};
```

Comment:
![image](https://user-images.githubusercontent.com/16799316/36402840-0812536e-15d9-11e8-806e-88a32be2bf97.png)
Entering an email which doesnt exist (not in DB) crashes the server as the it is not caught. Result here is an object (not null) is just empty object- You probably need to check the length after checking result isnt null